### PR TITLE
Add `actionview` bug report template

### PR DESCRIPTION
**Pull request by seanpdoyle**
### Motivation / Background

Introduce Action View bug report templates for contributors to reproduce issues with failing `ActionView::TestCase` instances.

### Detail

In addition to rendering ERB with the `inline:` keyword, the sample tests also include a `Helpers` module to demonstrate how to incorporate view helpers into the reproduction script.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.

### CODE REVIEW

File: guides/bug_report_templates/action_view_gem.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+  def test_stuff
+    view.extend Helpers
+
+    render inline: <<~ERB, locals: { key: "value" }
+      <p><%= upcase(key) %></p>
+    ERB
+
+    assert_pattern { rendered.html.at("p") => { content: "VALUE" } }
```
Comment (p8) [inline guides/bug_report_templates/action_view_gem.rb:31]: Pattern  matching doesn't work on 2.7, which is still supported by Rails 7.1.
```suggestion
    # pattern matching requires at least Ruby 3.0.
    assert_pattern { rendered.html.at("p") => { content: "VALUE" } }
```

File: guides/bug_report_templates/action_view_gem.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+class BugTest < ActionView::TestCase
+  module Helpers
+    def upcase(value)
+      value.upcase
+    end
+  end
+
+  def test_stuff
+    view.extend Helpers
```
Comment (rafaelfranca) [inline guides/bug_report_templates/action_view_gem.rb:24]: This isn't how you add helpers to test cases, you use the `helper` class method.

**Review by rafaelfranca — CHANGES_REQUESTED** [review]

File: guides/bug_report_templates/action_view_gem.rb (hunk 2 above)
Comment (seanpdoyle) [reply guides/bug_report_templates/action_view_gem.rb:24]: Thank you for calling that out! Is there somewhere in the documentation I could learn more about that? I can't find the method in [the API docs](https://api.rubyonrails.org/v7.1.1/classes/ActionView/TestCase.html)

Comment (rafaelfranca) [reply guides/bug_report_templates/action_view_gem.rb:24]: It is this method https://api.rubyonrails.org/classes/AbstractController/Helpers/ClassMethods.html#method-i-helper

Comment (seanpdoyle) [reply guides/bug_report_templates/action_view_gem.rb]: @rafaelfranca I've pushed up a change to use that method.

**Comment by zzak** [comment]
I'm starting to wonder if we really need both main && gem variations, when the only real difference is one line:

```diff
- gem "rails", github: "rails/rails", branch: "main"
+ gem "rails", "~> 7.1.0"
```

It could just be a comment, right?

**Comment by seanpdoyle** [comment]
@zzak @byroot I'v rebased this to account for the changes introduced in https://github.com/rails/rails/pull/50317.

**Comment by byroot** [comment]
CI failures seem unrelated.

**Pull request merged**